realtek: add support for hasivo F1100W-8S+ switch

koreapyj opened the pull request:
hasivo F1100W-8S+ switch is a 8 port unmanaged SFP+ switch.

Hardware:
- SoC: Realtek RTL9303 SoC
- RAM: DDR3 512MB
- Flash: Macronix MX25L25645G (SPI-NOR 32MB)
- Ethernet: 8xSFP+
- UART: [GND, RX, TX, 3.3V] (115200 8N1)

Known issues:
- Link status is always active
- 1G SFP modules not working
- No MAC addresses; this device was unmanaged switch

Notes:
- This PR includes LED behavior changes. It will move hardcoded led-nums into dts switch port nodes since this device has 3 LEDs but previous behavior cannot handle it correctly.

mans0n commented:
Thanks for the contribution, but I'd be glad if you can prepare your PR in more reviewable state. Your patches have formal issues (e.g. SoB, commented-out lines, broken indentation, ...) and lack proper explanation on the changes. You've already opened and abandoned a few draft PRs, and it makes reviewers less motivated to review your patches...

mans0n commented:
@bkobl Could you have a look for the rtl930x LED part?

koreapyj commented:
> Thanks for the contribution, but I'd be glad if you can prepare your PR in more reviewable state. Your patches have formal issues (e.g. SoB, commented-out lines, broken indentation, ...) and lack proper explanation on the changes. You've already opened and abandoned a few draft PRs, and it makes reviewers less motivated to review your patches...

Okay I'll aware it

koreapyj commented:
Device itself works correctly but forwarded traffics loss on default MTU (1500). It looks similar with my previous PR (https://github.com/openwrt/openwrt/pull/9888). Have any ideas?

koreapyj commented:
## GPIO
pin | description
----|---------------
0   | SFP 1 mod-def0
1   | SFP 2 mod-def0
2   | SFP 3 mod-def0
3   | SFP 4 mod-def0
4   | SFP 5 mod-def0
5   | SFP 6 mod-def0
6   | SFP 7 mod-def0
7   | SFP 8 mod-def0
8   | SFP * mod-def1
9   | SFP 1 mod-def2
10  | SFP 2 mod-def2
11  | SFP 3 mod-def2
12  | SFP 4 mod-def2
13  | SFP 5 mod-def2
14  | SFP 6 mod-def2
15  | SFP 7 mod-def2
16  | SFP 8 mod-def2
17  | 
18  | 
19  | 
20  | btn reset
21  | led sys
22  | 
23  | 

## Port LED
LED | descrption
-----|---------------
0   | red
1   | green
2   | SFP TX-disable?
3   |

bkobl commented on target/linux/realtek/dts-5.10/rtl9303_hasivo_f1100w-8splus.dts:
```diff
@@ -0,0 +1,434 @@
+	sfp0: sfp-p1 {
+		compatible = "sff,sfp";
+		i2c-bus = <&i2c01>;
+		los-gpio = <&gpio0 0 GPIO_ACTIVE_HIGH>;
```
You cannot have the same pin for module presence and for LOS. They have different meanings, one says the module is there, the other that the fiber has been unplugged. If you only have one of the 2, then only use the module presence pin.

koreapyj replied:
I've noticed it for a while but if I fix it sfp is probed and network goes broken. It is definitely a bug but at this point it'll break everything. So I'm going to have a look your repo and check it works with that.

oliv3r replied:
I don't think the kernel can claim 2 gpio's for one function. Do you have any idea how this is electrically connected?

If it is mod-def-0, then it is 'presence detect'.

bkobl commented:
Hi koreapyj!

This is impressive work. The hardware limitations such as no support for 1G and the link status should be gone with the latest developments here:
https://github.com/bkobl/openwrt/commits/rtl8214qf_merge/target/linux/realtek/files-5.10/drivers/net/dsa/rtl83xx
The issue is that some of the .dts settings would need to be adapted to work with the new code, e.g. the links would be "1000-basex", not 10GR, to be working with in-band management (1000-BaseX supports .3z in-band messaging, 10GR does not).

Adding a debugfs interface to the LED sets is a good idea, as it allows to test and change configuration at runtime, however, the defaults should probably be in the .dts. The working of the device should not depend on settings done via the debugfs. In principle the new code above has all the necessary code to make the LEDs work, a very similar device is: https://github.com/bkobl/openwrt/commit/672c05870e125a7e10a299700f213c633a464a8f

I would like to hear what @dangowrt has to say about this. Maybe we can coordinate these commits somehow into a PR focussing on getting the F1100W into master with all hardware features supported. For this one would need to cherry-pick the RTL9300 changes into your PR. Alternatively we get the F1100 in supported as is and fix things later.

Birger

koreapyj committed:
realtek: rtl930x: add debugfs for led

- Just for debugging
- For details, see register map

Signed-off-by: Yoonji Park <koreapyj@dcmys.kr>

koreapyj committed:
realtek: rtl930x: move hardcoded led-num to dts

- Issues on some devices

Signed-off-by: Yoonji Park <koreapyj@dcmys.kr>

koreapyj committed:
realtek: add support hasivo F1100W-8S+ switch
hasivo F1100W-8S+ switch is a 8 port SFP+ switch.

Hardware:
- SoC: Realtek RTL9303 SoC
- RAM: DDR3 512MB
- Flash: Macronix MX25L25645G (SPI-NOR 32MB)
- Ethernet: 8xSFP+
- UART: [GND, RX, TX, 3.3V] (115200 8N1)

Signed-off-by: Yoonji Park <koreapyj@dcmys.kr>

oliv3r commented:
@koreapyj I've added support based on your commit to my wip branch. Commit https://gitlab.com/olliver/openwrt/openwrt/-/commit/8cd26f48cc5fbc50c66362349509c16886994621 is the current state, It'll change over time for sure :)

http://www.hasivo.com/productinfo/895922.html is the switch this refers to, right?

So I'm a little confused on the led layout. You set it up as having 3 leds. I took the photo of that link above to determine that one must be orange (10/100/1000) and two for 10G (green). What I'm a little unsure of, why there's 3 led connections used. I guess one functions as sort of a directional thing? Idk, probably the hardware designers had no idea what they where doing, as with a rtl8231 can do 2 leds with just 2 wires, and now they are using 3.

Also, what's quite dumb, is that they are using gpio0 to check for mod-def-0, where that is the only pin that can be used as a dedicated LED driver. Sure, the GPIO led works. Sure, the designer must have been really proud he used sequential numbering. I suppose it's not so bad as the switch I'm working on that shifted the I2C bus by 1, meaning it is now a bit-banged bus, instead of using the hardware I2C peripherial. All just so that could keep their sorting of GPIO's nicely. Form over Function :)

Also, how much RAM does this switch have? I did a guestimate :p

Anyway, Feel free to test the dts, and mention any errors that you find.

koreapyj commented:
I am really glad to see your commits. But I'm busy now, so I'll definitely
check this one but I'll take some time. Please understand my delay. Thank
you so much.

2023년 3월 24일 (금) 오후 6:13, Olliver Schinagl ***@***.***>님이 작성:

> @koreapyj <https://github.com/koreapyj> I've added support based on your
> commit to my wip branch. Commit
> https://gitlab.com/olliver/openwrt/openwrt/-/commit/8cd26f48cc5fbc50c66362349509c16886994621
> is the current state, It'll change over time for sure :)
>
> http://www.hasivo.com/productinfo/895922.html is the switch this refers
> to, right?
>
> So I'm a little confused on the led layout. You set it up as having 3
> leds. I took the photo of that link above to determine that one must be
> orange (10/100/1000) and two for 10G (green). What I'm a little unsure of,
> why there's 3 led connections used. I guess one functions as sort of a
> directional thing? Idk, probably the hardware designers had no idea what
> they where doing, as with a rtl8231 can do 2 leds with just 2 wires, and
> now they are using 3.
>
> Also, what's quite dumb, is that they are using gpio0 to check for
> mod-def-0, where that is the only pin that can be used as a dedicated LED
> driver. Sure, the GPIO led works. Sure, the designer must have been really
> proud he used sequential numbering. I suppose it's not so bad as the switch
> I'm working on that shifted the I2C bus by 1, meaning it is now a
> bit-banged bus, instead of using the hardware I2C peripherial. All just so
> that could keep their sorting of GPIO's nicely. Form over Function :)
>
> Also, how much RAM does this switch have? I did a guestimate :p
>
> Anyway, Feel free to test the dts, and mention any errors that you find.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/openwrt/openwrt/pull/9950#issuecomment-1482481932>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAMYYG3S35GJ36JCCBIODBLW5VQUVANCNFSM5XFSP56Q>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

koreapyj commented:
> Would you be able to provide how you were able to obtain a serial console and share images of what points to solder to gain access to the serial console.

![image](https://user-images.githubusercontent.com/1674267/231360502-1a727298-801c-4390-a07d-3a70fb1a8c27.png)

Here you are

koreapyj commented:
[20090223085822.pdf](https://github.com/openwrt/openwrt/files/11222234/20090223085822.pdf)

RJ45 connector

koreapyj commented:
Some SFP modules works with uboot but not everything worked. 1G modules worked for me but 10G modules were not.

oliv3r commented:
> Would you be able to provide a dump of the SPI flash might have bricked mine

Technically, you don't really need it if your intend is using only openwrt. If you want to run vendor firmware, you can just install that instead. realtek sdk produces a 'bix' file, which can be flashed or booted, just like the openwrt initramfs image.

The biggest problem is the 'cfg' (and potentially OEM) partition(s). The cfg partition seems to hold mac addresses and such. Though with my openwrt patches, you don't really need to have it or put it there, its just some env variables, so if you can pull the mac from the sticker or some logs, you can restore it easily.

I'd try booting openwrt initramfs from the network; and see if you can recover anything at all first though ...

oliv3r commented:
> I am really glad to see your commits. But I'm busy now, so I'll definitely check this one but I'll take some time. Please understand my delay. Thank you so much. 2023년 3월 24일 (금) 오후 6:13, Olliver Schinagl ***@***.***>님이 작성:
> […](#)
> @koreapyj <https://github.com/koreapyj> I've added support based on your commit to my wip branch. Commit https://gitlab.com/olliver/openwrt/openwrt/-/commit/8cd26f48cc5fbc50c66362349509c16886994621 is the current state, It'll change over time for sure :) http://www.hasivo.com/productinfo/895922.html is the switch this refers to, right? So I'm a little confused on the led layout. You set it up as having 3 leds. I took the photo of that link above to determine that one must be orange (10/100/1000) and two for 10G (green). What I'm a little unsure of, why there's 3 led connections used. I guess one functions as sort of a directional thing? Idk, probably the hardware designers had no idea what they where doing, as with a rtl8231 can do 2 leds with just 2 wires, and now they are using 3. Also, what's quite dumb, is that they are using gpio0 to check for mod-def-0, where that is the only pin that can be used as a dedicated LED driver. Sure, the GPIO led works. Sure, the designer must have been really proud he used sequential numbering. I suppose it's not so bad as the switch I'm working on that shifted the I2C bus by 1, meaning it is now a bit-banged bus, instead of using the hardware I2C peripherial. All just so that could keep their sorting of GPIO's nicely. Form over Function :) Also, how much RAM does this switch have? I did a guestimate :p Anyway, Feel free to test the dts, and mention any errors that you find. — Reply to this email directly, view it on GitHub <[#9950 (comment)](https://github.com/openwrt/openwrt/pull/9950#issuecomment-1482481932)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AAMYYG3S35GJ36JCCBIODBLW5VQUVANCNFSM5XFSP56Q> . You are receiving this because you were mentioned.Message ID: ***@***.***>

No worries, just realize, atm, you are the only one that can validate and check if things are working :) I'll keep the patches in my tree and get them merged regardless though; but any feedback at some time is appreciated :)

koreapyj commented:
> Flashed realtek_sdk uboot and forgot to set the right board model so networking is not working

[s1100_8splus_fullflash.bin.gz](https://github.com/openwrt/openwrt/files/11237597/s1100_8splus_fullflash.bin.gz)

My full flash dump with factory firmware. Since this device was unmanaged switch so I think every device have same mac address 00::00.

oliv3r commented:
> Well progress kernel panic when booting @oliv3r fork of openwrt realtek-wip branch [boot.txt](https://github.com/openwrt/openwrt/files/11238048/boot.txt)

did you pull my latest WIP state? boots fine on an rtl9302b; and those crashes are unrelated to anything not SoC; e.g. not networking or anything.

It's failing to allocate the pinctrllers (which are the same on all rtl930x) and fails to init the serial port (identical on all realtek soc's afaik). Is the RAM amount correct? If its trying to allocate non-existant ram; then yeah, crashes would be expected :)

oliv3r commented:
> From which git repo branch and which site. Device has 128MB of ram
> 
> > > Well progress kernel panic when booting @oliv3r fork of openwrt realtek-wip branch [boot.txt](https://github.com/openwrt/openwrt/files/11238048/boot.txt)
> > 
> > 
> > did you pull my latest WIP state? boots fine on an rtl9302b; and those crashes are unrelated to anything not SoC; e.g. not networking or anything.
> > It's failing to allocate the pinctrllers (which are the same on all rtl930x) and fails to init the serial port (identical on all realtek soc's afaik). Is the RAM amount correct? If its trying to allocate non-existant ram; then yeah, crashes would be expected :)

I load the initramfs image as:
`rtk network on; tftpboot; bootm`
where my loadaddr is actualyl `loadaddr=0x84f00000` and i've set `bootfile` to whatever file you are using.

Btw, here's my bootlog;

<details>

<summary>full bootlog</summary>

```
U-Boot 2011.12.001-owrt (Aug 12 2022 - 14:40:05)

Board: RTL9300 CPU:800MHz LX:175MHz DDR:600MHz
DRAM:  128 MB
SPI-F: MXIC/C22018/MMIO16-1/ModeC 1x16 MB (plr_flash_info @ 83f5a148)
Loading 65536B env. variables from offset 0xe0000
Net:   Net Initialization Skipped
No ethernet found.
Hit Esc key to stop autoboot:  0 
Enable network
RTCORE Driver Module Initialize
  IOAL init
  Hardware-profile probe    (Forced profile: RTL9302B_8218D_2x8226_2xGE)
 (RTL9302B_8218D_2x8226_2xGE)
  Hardware-profile init
  GPIO probe (unit 0): (found)
  GPIO Init
Reset PHY gpio OK!
  SPI init (unit 0) (type3)
    SPI Master init
  I2C probe (unit 0)
  I2C init (unit 0)
  NIC probe (unit 0)
  Loader RTNIC Driver Module Initialize
  IOAL init
RTK Driver Module Initialize
  MAC probe (unit 0)
    Chip 9302 (found)
  MAC init (unit 0)
  SMI protocol probe (unit 0)
  PHY probe (unit 0)
  Chip Construct (unit 0)
    Chip Construct
    Disable PHY Polling
    PHY Reset
    MAC Construct
    Turn Off Serdes
    Serdes Construct
    PHY Construct
    Turn On Serdes
    Enable PHY Polling
    Misc
  PHY init (unit 0)
  Mgmt_dev init (unit 0) 
Please wait for PHY init-time ...

Using rtl9300#0 device
TFTP from server 192.168.163.1; our IP address is 192.168.163.10
Filename 'openwrt-realtek-otto-24kc-zyxel_xgs1010-12_vendor-initramfs-kernel.bin'.
Load address: 0x84f00000
Loading: T #################################################################
	 #################################################################
	 #################################################################
	 #################################################################
	 #################################################################
	 #####
done
Bytes transferred = 4831275 (49b82b hex)
## Booting kernel from Legacy Image at 84f00000 ...
   Image Name:   MIPS OpenWrt Linux-5.15.106
   Created:      2023-02-16  17:00:03 UTC
   Image Type:   MIPS Linux Kernel Image (lzma compressed)
   Data Size:    4831211 Bytes = 4.6 MB
   Load Address: 80100000
   Entry Point:  80100000
   Verifying Checksum ... OK
   Uncompressing Kernel Image ... OK

Starting kernel ...

[    0.000000] Linux version 5.15.106 (buildbot@3d06adae80e2) (mips-openwrt-linux-musl-gcc (OpenWrt GCC 12.2.0 unknown) 12.2.0, GNU ld (GNU Binutils) 2.40.0) #0 SMP Thu Feb 16 17:00:03 2023
[    0.000000] printk: bootconsole [early0] enabled
[    0.000000] CPU0 revision is: 00019555 (MIPS 34Kc)
[    0.000000] SoC: Realtek RTL9302B v1 ([0x93021001] variant: 10G VER: 1 MCID: 0x0 ID: 0x6487)
[    0.000000] MIPS: machine is Zyxel XGS1010-12 Switch (vendor)
[    0.000000] earlycon: ns16550a0 at MMIO 0x18002000 (options '115200n8')
[    0.000000] printk: bootconsole [ns16550a0] enabled
[    0.000000] Initrd not found or empty - disabling initrd
[    0.000000] Calling board specific registration of SMP ops
[    0.000000] Detected 1 available secondary CPU(s)
[    0.000000] Primary instruction cache 32kB, VIPT, 4-way, linesize 32 bytes.
[    0.000000] Primary data cache 32kB, 4-way, PIPT, no aliases, linesize 32 bytes
[    0.000000] Zone ranges:
[    0.000000]   Normal   [mem 0x0000000000000000-0x0000000007ffffff]
[    0.000000] Movable zone start for each node
[    0.000000] Early memory node ranges
[    0.000000]   node   0: [mem 0x0000000000000000-0x0000000007ffffff]
[    0.000000] Initmem setup node 0 [mem 0x0000000000000000-0x0000000007ffffff]
[    0.000000] percpu: Embedded 11 pages/cpu s15760 r8192 d21104 u45056
[    0.000000] pcpu-alloc: s15760 r8192 d21104 u45056 alloc=11*4096
[    0.000000] pcpu-alloc: [0] 0 [0] 1 
[    0.000000] Built 1 zonelists, mobility grouping on.  Total pages: 32480
[    0.000000] Kernel command line: earlycon ip=dhcp
[    0.000000] Unknown kernel command line parameters "ip=dhcp", will be passed to user space.
[    0.000000] Dentry cache hash table entries: 16384 (order: 4, 65536 bytes, linear)
[    0.000000] Inode-cache hash table entries: 8192 (order: 3, 32768 bytes, linear)
[    0.000000] Writing ErrCtl register=0003486b
[    0.000000] Readback ErrCtl register=0003486b
[    0.000000] mem auto-init: stack:off, heap alloc:off, heap free:off
[    0.000000] Memory: 110356K/131072K available (6544K kernel code, 652K rwdata, 1380K rodata, 10488K init, 214K bss, 20716K reserved, 0K cma-reserved)
[    0.000000] SLUB: HWalign=128, Order=0-3, MinObjects=0, CPUs=2, Nodes=1
[    0.000000] rcu: Hierarchical RCU implementation.
[    0.000000] 	Tracing variant of Tasks RCU enabled.
[    0.000000] rcu: RCU calculated value of scheduler-enlistment delay is 10 jiffies.
[    0.000000] NR_IRQS: 256
[    0.000000] rtl83xx-clk: initialized, SW 2450 MHz, CPU 800 MHz, MEM 600 MHz, LXB 175 MHz
[    0.000000] clocksource: realtek_otto_timer: mask: 0xfffffff max_cycles: 0xfffffff, max_idle_ns: 38225208801 ns
[    0.000001] sched_clock: 28 bits at 3125kHz, resolution 320ns, wraps every 42949672800ns
[    0.008959] Failed to request irq 7 (timer)
[    0.013510] clocksource: MIPS: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 4778151116 ns
[    0.023279] sched_clock: 32 bits at 400MHz, resolution 2ns, wraps every 5368709118ns
[    0.031879] Calibrating delay loop... 531.66 BogoMIPS (lpj=2658304)
[    0.095701] pid_max: default: 32768 minimum: 301
[    0.101010] Mount-cache hash table entries: 1024 (order: 0, 4096 bytes, linear)
[    0.109050] Mountpoint-cache hash table entries: 1024 (order: 0, 4096 bytes, linear)
[    0.121794] rcu: Hierarchical SRCU implementation.
[    0.127582] dyndbg: Ignore empty _ddebug table in a CONFIG_DYNAMIC_DEBUG_CORE build
[    0.136498] smp: Bringing up secondary CPUs ...
[    0.142906] Primary instruction cache 32kB, VIPT, 4-way, linesize 32 bytes.
[    0.142942] Primary data cache 32kB, 4-way, PIPT, no aliases, linesize 32 bytes
[    0.143032] CPU1 revision is: 00019555 (MIPS 34Kc)
[    0.198945] Synchronize counters for CPU 1: done.
[    0.225103] smp: Brought up 1 node, 2 CPUs
[    0.234561] clocksource: jiffies: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 19112604462750000 ns
[    0.245346] futex hash table entries: 512 (order: 4, 65536 bytes, linear)
[    0.253008] pinctrl core: initialized pinctrl subsystem
[    0.259968] NET: Registered PF_NETLINK/PF_ROUTE protocol family
[    0.271709] FPU Affinity set after 10620 emulations
[    0.297631] rtl83xx-clk lexra-busp:clock-controller: rate setting enabled, SW 2450-2450 MHz, CPU 400-975 MHz, MEM 600-600 MHz, LXB 175-175 MHz, OVERCLOCK AT OWN RISK
[    0.316324] clocksource: Switched to clocksource realtek_otto_timer
[    0.324441] NET: Registered PF_INET protocol family
[    0.330120] IP idents hash table entries: 2048 (order: 2, 16384 bytes, linear)
[    0.338970] tcp_listen_portaddr_hash hash table entries: 512 (order: 0, 6144 bytes, linear)
[    0.348191] Table-perturb hash table entries: 65536 (order: 6, 262144 bytes, linear)
[    0.356727] TCP established hash table entries: 1024 (order: 0, 4096 bytes, linear)
[    0.365100] TCP bind hash table entries: 1024 (order: 1, 8192 bytes, linear)
[    0.372874] TCP: Hash tables configured (established 1024 bind 1024)
[    0.380015] UDP hash table entries: 256 (order: 1, 8192 bytes, linear)
[    0.387274] UDP-Lite hash table entries: 256 (order: 1, 8192 bytes, linear)
[    0.395271] NET: Registered PF_UNIX/PF_LOCAL protocol family
[    0.407107] workingset: timestamp_bits=14 max_order=15 bucket_order=1
[    0.422049] squashfs: version 4.0 (2009/01/31) Phillip Lougher
[    0.428588] jffs2: version 2.2 (NAND) (SUMMARY) (ZLIB) (LZMA) (RTIME) (CMODE_PRIORITY) (c) 2001-2006 Red Hat, Inc.
[    0.444908] pinctrl-single 1b00cc00.pinmux: 32 pins, size 4
[    0.452013] pinctrl-single 1b000384.pinmux: 32 pins, size 4
[    0.543425] Serial: 8250/16550 driver, 2 ports, IRQ sharing disabled
[    0.552614] printk: console [ttyS0] disabled
[    0.557633] 18002000.uart: ttyS0 at MMIO 0x18002000 (irq = 30, base_baud = 10937500) is a 16550A
[    0.567469] printk: console [ttyS0] enabled
[    0.567469] printk: console [ttyS0] enabled
[    0.576961] printk: bootconsole [early0] disabled
[    0.576961] printk: bootconsole [early0] disabled
[    0.587582] printk: bootconsole [ns16550a0] disabled
[    0.587582] printk: bootconsole [ns16550a0] disabled
[    0.683150] brd: module loaded
[    0.698390] spi-nor spi0.0: mx25l12805d (16384 Kbytes)
[    0.704266] 8 fixed-partitions partitions found on MTD device spi0.0
[    0.711608] Creating 8 MTD partitions on "spi0.0":
[    0.717121] 0x000000000000-0x0000000e0000 : "u-boot"
[    0.725173] 0x0000000e0000-0x0000000f0000 : "u-boot-env"
[    0.733732] 0x0000000f0000-0x000000100000 : "u-boot-sysenv"
[    0.742673] 0x000000100000-0x000000810000 : "_firmware_concat0"
[    0.751751] 0x000000810000-0x000000820000 : "HTP_logs"
[    0.760639] 0x000000820000-0x0000008f0000 : "_firmware_concat1"
[    0.769898] 0x0000008f0000-0x000000900000 : "flash_write_test"
[    0.779486] 0x000000900000-0x000001000000 : "_firmware_concat2"
[    0.909557] Probing RTL838X eth device pdev: 8208c400, dev: 8208c410
[    0.980499] Found SoC ID: 9302: RTL9302B v1 ([0x93021001] variant: 10G VER: 1 MCID: 0x0 ID: 0x6487), family 9300
[    0.992441] Using MAC 000000e04c000000
[    1.299808] RTL8226-CG 2.5Gbps PHY: probe of mdio-bus:18 failed with error -5
[    1.398507] RTL8226-CG 2.5Gbps PHY: probe of mdio-bus:19 failed with error -5
[    1.691687] REALTEK RTL9300 SERDES mdio-bus:1a: Detected internal RTL9300 Serdes
[    1.984493] REALTEK RTL9300 SERDES mdio-bus:1b: Detected internal RTL9300 Serdes
[    1.999261] i2c_dev: i2c /dev entries driver
[    2.007232] i2c i2c-0: Added multiplexed i2c bus 1
[    2.013223] i2c i2c-0: Added multiplexed i2c bus 2
[    2.018688] i2c-mux-regmap lexra-bus0:i2c-mux@0: Registered regmap I2C mux
[    2.027454] i2c-mux-regmap lexra-bus0:i2c-mux@1: Failed to register regmap I2C mux
[    2.039423] realtek-crypto lexra-bus0:crypto@c000: no IO address given
[    2.048383] NET: Registered PF_INET6 protocol family
[    2.068143] Segment Routing with IPv6
[    2.072335] In-situ OAM (IOAM) with IPv6
[    2.076923] NET: Registered PF_PACKET protocol family
[    2.083128] 8021q: 802.1Q VLAN Support v1.8
[    2.089411] Concatenating MTD devices:
[    2.093597] (0): "_firmware_concat0"
[    2.097796] (1): "_firmware_concat1"
[    2.101791] (2): "_firmware_concat2"
[    2.105758] into device "virtual_flash"
[    2.110115] 1 fixed-partitions partitions found on MTD device virtual_flash
[    2.118383] Creating 1 MTD partitions on "virtual_flash":
[    2.124476] 0x000000000000-0x000000ee0000 : "firmware"
[    2.163444] sfp sfp-p11: Host maximum power 1.0W
[    2.171921] sfp sfp-p12: Host maximum power 1.0W
[    2.945645] REALTEK RTL9300 SERDES rtl838x slave mii-0:1a: Detected internal RTL9300 Serdes
[    2.989887] REALTEK RTL9300 SERDES rtl838x slave mii-0:1b: Detected internal RTL9300 Serdes
[    3.526438] random: crng init done
[    3.763173] sfp sfp-p11: EEPROM base structure checksum failure: 0xc1 != 0xff
[    3.771332] sfp EE: 00000000: ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff  ................
[    3.781133] sfp EE: 00000010: ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff  ................
[    3.790916] sfp EE: 00000020: ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff  ................
[    3.800789] sfp EE: 00000030: ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff ff  ................
[    3.810616] sfp EE: 00000040: 00 00 00 00 82 45 9a 00 82 7b 7e 7c 80 8d 00 00  .....E...{~|....
[    3.820462] sfp EE: 00000050: 81 50 39 00 81 50 39 00 80 15 de 30 82 16 8b 00  .P9..P9....0....
[    3.918368] REALTEK RTL9300 SERDES rtl838x slave mii-0:3f: Detected internal RTL9300 Serdes
[    3.949072] rtl93xx_setup called
[    3.952719] In rtl83xx_vlan_setup
[    3.956469] In rtl930x_vlan_profile_setup
[    3.960961] In rtl930x_vlan_profile_setup
[    3.965421] UNKNOWN_MC_PMASK: 000000001fffffff
[    3.970463] VLAN 0: L2 learn: 0; Unknown MC PMasks: L2 1fffffff, IPv4 1fffffff, IPv6: 1fffffff
[    3.970491]   Routing enabled: IPv4 UC y, IPv6 UC y, IPv4 MC y, IPv6 MC y
[    3.987681]   Bridge enabled: IPv4 MC n, IPv6 MC n,
[    3.993117] VLAN profile 0: raw 00033000 00000000 1fffffff 1fffffff 1fffffff
[    5.066381] ################# rtl83xx_enable_phy_polling:          f0000ff
[    5.074062] rtl930x_pie_init
[    5.077524] rtl930x_led_init called
[    5.081564] rtl930x_led_init bb00cc00: 01052659
[    5.086651] rtl930x_led_init bb00cc04: 00005555
[    5.091699] rtl930x_led_init bb00cc08: 005f0000
[    5.096775] rtl930x_led_init bb00cc0c: 00000000
[    5.101819] rtl930x_led_init bb00cc10: 0a200a01
[    5.106895] rtl930x_led_init bb00cc14: 0a0b0a28
[    5.111939] rtl930x_led_init bb00cc18: 0a820a0b
[    5.117014] rtl930x_led_init bb00cc1c: 0b800a20
[    5.122059] rtl930x_led_init bb00cc20: 0a080a08
[    5.127152] rtl930x_led_init bb00cc24: 00000000
[    5.132196] rtl930x_led_init bb00cc28: 0a300bc0
[    5.137273] rtl930x_led_init bb00cc2c: 00000000
[    5.142316] rtl930x_led_init bb00cc30: 00f50000
[    5.147418] rtl930x_led_init bb00cc34: 00000000
[    5.152472] rtl930x_led_init bb00cc38: 00f50000
[    5.157553] rtl930x_led_init bb00cc3c: 030000ff
[    5.162600] rtl930x_led_init bb00cc40: 0c000000
[    5.167680] rtl930x_led_init bb00cc44: 0f0000ff
[    5.172730] rtl930x_led_init bb00cc48: 00000000
[    5.177822] rtl930x_led_init bb00cc4c: 00000000
[    5.182876] rtl930x_led_init bb00cc50: 00000000
[    5.187975] rtl930x_led_init bb00cc54: 00000000
[    5.193023] rtl930x_led_init bb00cc58: 00000000
[    5.198102] rtl930x_led_init bb00cc5c: 00000000
[    5.203663] rtl83xx-switch lexra-bus0:switch@1b000000: configuring for fixed/internal link mode
[    5.213468] rtl930x_phylink_mac_config port 28, mode 1, phy-mode: internal, speed 1000, link 0
[    5.223287] rtl9300_configure_8218d: A Port 0, SerDes is 2 phy-mode xgmii port 0
[    5.231720] rtl930x_phylink_mac_config port 28, mode 1, phy-mode: internal, speed 1000, link 1
[    5.231850] rtl9300_configure_8218d CMU BAND is 16
[    5.241460] rtl83xx-switch lexra-bus0:switch@1b000000: Link is Up - 1Gbps/Full - flow control off
[    5.286396] rtl9300_sds_set 31
[    5.289829] rtl9300_configure_8218d PATCHING SerDes 2
[    5.296938] rtl9300_configure_8218d: Configuring RTL9300 SERDES 2, mode 10
[    5.307539] rtl8218d_serdes_mode_get, reading phy 0 got me block 0000 and data 00b0
[    5.316080] rtl8218d_serdes_mode_get: SDS mode: b0
[    5.324396] rtl8218d_serdes_model_get: 8281D model: 81
[    5.330243] rtl9300_serdes_mac_link_config: registers before 00000000 00001403
[    5.338452] rtl9300_serdes_mac_link_config: registers after 00000000 00001403
[    5.346446] rtl9300_sds_set 16
[    5.352879] rtl83xx-switch lexra-bus0:switch@1b000000 lan1 (uninitialized): PHY [mdio-bus:00] driver [REALTEK RTL8218D] (irq=POLL)
[    5.368214] rtl83xx-switch lexra-bus0:switch@1b000000 lan2 (uninitialized): PHY [mdio-bus:01] driver [REALTEK RTL8218D] (irq=POLL)
[    5.384097] rtl83xx-switch lexra-bus0:switch@1b000000 lan3 (uninitialized): PHY [mdio-bus:02] driver [REALTEK RTL8218D] (irq=POLL)
[    5.399371] rtl83xx-switch lexra-bus0:switch@1b000000 lan4 (uninitialized): PHY [mdio-bus:03] driver [REALTEK RTL8218D] (irq=POLL)
[    5.415339] rtl83xx-switch lexra-bus0:switch@1b000000 lan5 (uninitialized): PHY [mdio-bus:04] driver [REALTEK RTL8218D] (irq=POLL)
[    5.430679] rtl83xx-switch lexra-bus0:switch@1b000000 lan6 (uninitialized): PHY [mdio-bus:05] driver [REALTEK RTL8218D] (irq=POLL)
[    5.446599] rtl83xx-switch lexra-bus0:switch@1b000000 lan7 (uninitialized): PHY [mdio-bus:06] driver [REALTEK RTL8218D] (irq=POLL)
[    5.462584] rtl83xx-switch lexra-bus0:switch@1b000000 lan8 (uninitialized): PHY [mdio-bus:07] driver [REALTEK RTL8218D] (irq=POLL)
[    5.477830] rtl9300_configure_rtl8226 configuring RTL8226 on port 24
[    5.484975] rtl9300_configure_rtl8226: port 24, SerDes is 6
[    5.491431] rtl9300_configure_rtl8226 CMU BAND is 16
[    5.536381] rtl9300_force_sds_mode: SDS: 6, mode 0
[    5.541732] rtl9300_force_sds_mode --------------------- serdes 6 forcing to 1f ...
[    5.550508] rtl9300_configure_rtl8226 PATCHING SerDes 6
[    5.560468] rtl9300_rtl8226_phy_setup, port 24 patch version 15
[    5.567125] rtl9300_rtl8226_phy_setup: MDI pins already swapped
[    5.573783] rtl9300_serdes_mac_link_config: registers before 00000000 00001403
[    5.581983] rtl9300_serdes_mac_link_config: registers after 00000000 00001403
[    5.592644] rtl9300_force_sds_mode: SDS: 6, mode 18
[    5.598139] rtl9300_force_sds_mode --------------------- serdes 6 forcing to 12 ...
[    5.826345] rtl9300_force_sds_mode toggling LC or Ring for 10gr, round 0
[    5.886459] rtl9300_force_sds_mode end power 0x20 0 30
[    5.892194] rtl9300_force_sds_mode --------------------- serdes 6 forced to 12 DONE
[    5.901143] rtl83xx-switch lexra-bus0:switch@1b000000 lan9 (uninitialized): PHY [mdio-bus:18] driver [REALTEK RTL8226] (irq=POLL)
[    5.916105] rtl9300_configure_rtl8226 configuring RTL8226 on port 25
[    5.923336] rtl9300_configure_rtl8226: port 25, SerDes is 7
[    5.929796] rtl9300_configure_rtl8226 CMU BAND is 16
[    5.966384] rtl9300_force_sds_mode: SDS: 7, mode 0
[    5.971741] rtl9300_force_sds_mode --------------------- serdes 7 forcing to 1f ...
[    5.980494] rtl9300_configure_rtl8226 PATCHING SerDes 7
[    5.989991] rtl9300_rtl8226_phy_setup, port 25 patch version 15
[    5.996651] rtl9300_rtl8226_phy_setup: MDI pins already swapped
[    6.003310] rtl9300_serdes_mac_link_config: registers before 00000000 00001403
[    6.011512] rtl9300_serdes_mac_link_config: registers after 00000000 00001403
[    6.022183] rtl9300_force_sds_mode: SDS: 7, mode 18
[    6.027673] rtl9300_force_sds_mode --------------------- serdes 7 forcing to 12 ...
[    6.256345] rtl9300_force_sds_mode toggling LC or Ring for 10gr, round 0
[    6.316458] rtl9300_force_sds_mode end power 0x20 0 30
[    6.322196] rtl9300_force_sds_mode --------------------- serdes 7 forced to 12 DONE
[    6.331155] rtl83xx-switch lexra-bus0:switch@1b000000 lan10 (uninitialized): PHY [mdio-bus:19] driver [REALTEK RTL8226] (irq=POLL)
[    6.350481] DSA: tree 0 setup
[    6.353913] LINK state irq: 23
[    6.357445] rtl930x_switch_irq link faults: 0fffff00
[    6.357507] In rtl83xx_setup_qos
[    6.362999] rtl930x_switch_irq RX symbol errors: 00000004
[    6.366708] L3_IPUC_ROUTE_CTRL 00002000, IPMC_ROUTE 0000077e, IP6UC_ROUTE 00014400, IP6MC_ROUTE 0001db7e
[    6.372575] rtl930x_switch_irq link faults: 0fffff00
[    6.372587] rtl930x_switch_irq RX symbol errors: 00000004
[    6.372602] rtl930x_switch_irq link faults: 0fffff00
[    6.372613] rtl930x_switch_irq RX symbol errors: 00000004
[    6.372633] rtl930x_switch_irq link faults: 0fffff00
[    6.372643] rtl930x_switch_irq RX symbol errors: 00000004
[    6.417893] L3_IPUC_ROUTE_CTRL 00002001, IPMC_ROUTE 00000501, IP6UC_ROUTE 00014581, IP6MC_ROUTE 00012881
[    6.428493] L3_IP_ROUTE_CTRL 0000013f
[    6.432684] rtl930x_dbgfs_init called
[    6.436874] rtl83xx_fib_event_work_do: FIB4 default rule failed
[    6.443477] rtl83xx_fib_event_work_do: FIB4 default rule failed
[    6.447224] i2c-mux-regmap lexra-bus0:i2c-mux@1: Failed to register regmap I2C mux
[    6.459265] i2c-mux-regmap lexra-bus0:i2c-mux@1: Failed to register regmap I2C mux
[    6.513859] Freeing unused kernel image (initmem) memory: 10488K
[    6.520637] This architecture does not have kernel memory protection.
[    6.527868] Run /init as init process
[    6.531949]   with arguments:
[    6.535231]     /init
[    6.537791]   with environment:
[    6.541297]     HOME=/
[    6.543912]     TERM=linux
[    6.546944]     ip=dhcp
[    7.116504] init: Console is alive
[    7.120821] init: - watchdog -
[    7.144986] kmodloader: loading kernel modules from /etc/modules-boot.d/*
[    7.155664] kmodloader: done loading kernel modules from /etc/modules-boot.d/*
[    7.166114] init: - preinit -
[    8.685098] RESETTING 9300, CPU_PORT 28
[    8.928513] rtl838x-eth 1b00a300.ethernet eth0: configuring for fixed/internal link mode
[    8.937608] In rtl838x_mac_config, mode 1
[    8.943706] rtl83xx-switch lexra-bus0:switch@1b000000 lan1: configuring for phy/xgmii link mode
[    8.953448] rtl930x_phylink_mac_config port 0, mode 0, phy-mode: xgmii, speed -1, link 0
[    8.962502] rtl930x_phylink_mac_config SDS is 2
[    8.967596] rtl930x_phylink_mac_config: Unsupported speed: -1
[    8.974147] In rtl838x_mac_link_up, mode 1
[    8.974409] 8021q: adding VLAN 0 to HW filter on device lan1
[    8.978854] rtl838x-eth 1b00a300.ethernet eth0: Link is Up - 1Gbps/Full - flow control off
[    8.994470] IPv6: ADDRCONF(NETDEV_CHANGE): eth0: link becomes ready
[    9.001637] rtl83xx_fib_event: FIB_RULE ADD/DEL for IPv6 not supported
[    9.009201] rtl83xx_fib_event: FIB_RULE ADD/DEL for IPv6 not supported
[    9.017867] rtl83xx_fib_event_work_do: FIB4 failed
[    9.023225] rtl83xx_fib_event_work_do: FIB4 failed
[    9.028755] rtl83xx_fib_event_work_do: FIB4 failed
Press the [f] key and hit [enter] to enter failsafe mode
Press the [1], [2], [3] or [4] key and hit [enter] to select the debug level
[   10.586399] rtl83xx_fib_event: FIB_RULE ADD/DEL for IPv6 not supported
[   11.177625] rtl83xx_fib4_del: no such gateway: 0.0.0.0
[   11.183379] rtl83xx_fib4_del: no such gateway: 0.0.0.0
[   11.189320] rtl83xx_fib4_del: no such gateway: 0.0.0.0
[   11.217810] procd: - early -
[   11.221390] procd: - watchdog -
[   12.538704] procd: - watchdog -
[   12.542805] procd: - ubus -
[   12.600889] procd: - init -
Please press Enter to activate this console.
[   13.184928] kmodloader: loading kernel modules from /etc/modules.d/*
[   13.204237] GACT probability on
[   13.210987] Mirror/redirect action on
[   13.230304] u32 classifier
[   13.233337]     input device check on
[   13.237555]     Actions configured
[   13.355279] kmodloader: done loading kernel modules from /etc/modules.d/*
[   13.451066] urngd: v1.0.2 started.
[   16.473893] rtl83xx_fib_event: FIB_RULE ADD/DEL for IPv6 not supported
[   27.928046] rtl83xx_fib_event: FIB_RULE ADD/DEL for IPv6 not supported
[   27.970581] in rtl838x_eth_stop
[   27.974188] In rtl838x_mac_link_down, mode 1
[   27.979168] rtl838x-eth 1b00a300.ethernet eth0: Link is Down
[   28.546734] rtl83xx_fib_event: FIB_RULE ADD/DEL for IPv6 not supported
[   28.554152] rtl83xx_fib_event: FIB_RULE ADD/DEL for IPv6 not supported
[   28.561644] rtl83xx_fib_event: FIB_RULE ADD/DEL for IPv6 not supported
[   28.569087] rtl83xx_fib_event: FIB_RULE ADD/DEL for IPv6 not supported
[   28.605200] Using MAC 000000004c000000
[   28.612166] RESETTING 9300, CPU_PORT 28
[   28.858928] rtl838x-eth 1b00a300.ethernet eth0: configuring for fixed/internal link mode
[   28.868057] In rtl838x_mac_config, mode 1
[   28.873033] In rtl838x_mac_link_up, mode 1
[   28.877829] rtl838x-eth 1b00a300.ethernet eth0: Link is Up - 1Gbps/Full - flow control off
[   28.887285] IPv6: ADDRCONF(NETDEV_CHANGE): eth0: link becomes ready
[   28.894417] rtl83xx_fib_event: FIB_RULE ADD/DEL for IPv6 not supported
[   28.902027] rtl83xx_fib_event: FIB_RULE ADD/DEL for IPv6 not supported
[   28.915065] device eth0 entered promiscuous mode
[   28.921785] rtl83xx-switch lexra-bus0:switch@1b000000 lan1: configuring for phy/xgmii link mode
[   28.931623] rtl930x_phylink_mac_config port 0, mode 0, phy-mode: xgmii, speed -1, link 0
[   28.940766] rtl930x_phylink_mac_config SDS is 2
[   28.945840] rtl930x_phylink_mac_config: Unsupported speed: -1
[   28.953938] 8021q: adding VLAN 0 to HW filter on device lan1
[   28.968930] switch0: port 1(lan1) entered blocking state
[   28.974865] switch0: port 1(lan1) entered disabled state
[   28.982131] device lan1 entered promiscuous mode
[   29.019079] device switch0 entered promiscuous mode
[   29.057006] rtl83xx-switch lexra-bus0:switch@1b000000 lan2: configuring for phy/xgmii link mode
[   29.066872] rtl930x_phylink_mac_config port 1, mode 0, phy-mode: xgmii, speed -1, link 0
[   29.075923] rtl930x_phylink_mac_config SDS is -1
[   29.081212] rtl930x_phylink_mac_config: Unsupported speed: -1
[   29.091131] 8021q: adding VLAN 0 to HW filter on device lan2
[   29.117779] switch0: port 2(lan2) entered blocking state
[   29.123773] switch0: port 2(lan2) entered disabled state
[   29.137938] device lan2 entered promiscuous mode
[   29.191972] rtl83xx-switch lexra-bus0:switch@1b000000 lan3: configuring for phy/xgmii link mode
[   29.201819] rtl930x_phylink_mac_config port 2, mode 0, phy-mode: xgmii, speed -1, link 0
[   29.211092] rtl930x_phylink_mac_config SDS is -1
[   29.216380] rtl930x_phylink_mac_config: Unsupported speed: -1
[   29.242271] 8021q: adding VLAN 0 to HW filter on device lan3
[   29.258323] switch0: port 3(lan3) entered blocking state
[   29.264296] switch0: port 3(lan3) entered disabled state
[   29.271817] device lan3 entered promiscuous mode
[   29.288761] rtl83xx-switch lexra-bus0:switch@1b000000 lan4: configuring for phy/xgmii link mode
[   29.298535] rtl930x_phylink_mac_config port 3, mode 0, phy-mode: xgmii, speed -1, link 0
[   29.307693] rtl930x_phylink_mac_config SDS is -1
[   29.312863] rtl930x_phylink_mac_config: Unsupported speed: -1
[   29.320149] 8021q: adding VLAN 0 to HW filter on device lan4
[   29.335895] switch0: port 4(lan4) entered blocking state
[   29.342007] switch0: port 4(lan4) entered disabled state
[   29.349122] device lan4 entered promiscuous mode
[   29.365998] rtl83xx-switch lexra-bus0:switch@1b000000 lan5: configuring for phy/xgmii link mode
[   29.375758] rtl930x_phylink_mac_config port 4, mode 0, phy-mode: xgmii, speed -1, link 0
[   29.384858] rtl930x_phylink_mac_config SDS is -1
[   29.390097] rtl930x_phylink_mac_config: Unsupported speed: -1
[   29.396937] 8021q: adding VLAN 0 to HW filter on device lan5
[   29.412554] switch0: port 5(lan5) entered blocking state
[   29.418562] switch0: port 5(lan5) entered disabled state
[   29.425508] device lan5 entered promiscuous mode
[   29.442652] rtl83xx-switch lexra-bus0:switch@1b000000 lan6: configuring for phy/xgmii link mode
[   29.452410] rtl930x_phylink_mac_config port 5, mode 0, phy-mode: xgmii, speed -1, link 0
[   29.461508] rtl930x_phylink_mac_config SDS is -1
[   29.466778] rtl930x_phylink_mac_config: Unsupported speed: -1
[   29.473565] 8021q: adding VLAN 0 to HW filter on device lan6
[   29.489307] switch0: port 6(lan6) entered blocking state
[   29.495244] switch0: port 6(lan6) entered disabled state
[   29.502473] device lan6 entered promiscuous mode
[   29.519817] rtl83xx-switch lexra-bus0:switch@1b000000 lan7: configuring for phy/xgmii link mode
[   29.529590] rtl930x_phylink_mac_config port 6, mode 0, phy-mode: xgmii, speed -1, link 0
[   29.538726] rtl930x_phylink_mac_config SDS is -1
[   29.543896] rtl930x_phylink_mac_config: Unsupported speed: -1
[   29.551022] 8021q: adding VLAN 0 to HW filter on device lan7
[   29.566711] switch0: port 7(lan7) entered blocking state
[   29.572651] switch0: port 7(lan7) entered disabled state
[   29.579887] device lan7 entered promiscuous mode
[   29.596597] rtl83xx-switch lexra-bus0:switch@1b000000 lan8: configuring for phy/xgmii link mode
[   29.606364] rtl930x_phylink_mac_config port 7, mode 0, phy-mode: xgmii, speed -1, link 0
[   29.615392] rtl930x_phylink_mac_config SDS is -1
[   29.620606] rtl930x_phylink_mac_config: Unsupported speed: -1
[   29.627676] 8021q: adding VLAN 0 to HW filter on device lan8
[   29.643408] switch0: port 8(lan8) entered blocking state
[   29.649425] switch0: port 8(lan8) entered disabled state
[   29.656545] device lan8 entered promiscuous mode
[   29.672147] rtl83xx-switch lexra-bus0:switch@1b000000 lan9: configuring for phy/hsgmii link mode
[   29.682011] rtl930x_phylink_mac_config port 24, mode 0, phy-mode: hsgmii, speed -1, link 0
[   29.691333] rtl930x_phylink_mac_config SDS is 6
[   29.696507] rtl930x_phylink_mac_config: Unsupported speed: -1
[   29.702925] rtl930x_phylink_mac_config CURRENT FORCED MODE 18
[   29.709387] rtl9300_rtl8226_mode_set setting serdes 6 to mode sgmii +++++
[   29.756414] rtl9300_force_sds_mode: SDS: 6, mode 0
[   29.761768] rtl9300_force_sds_mode --------------------- serdes 6 forcing to 1f ...
[   29.770648] rtl9300_serdes_mac_link_config: registers before 00000000 00001403
[   29.778940] rtl9300_serdes_mac_link_config: registers after 00000000 00001403
[   29.789802] rtl9300_force_sds_mode: SDS: 6, mode 4
[   29.795155] rtl9300_force_sds_mode --------------------- serdes 6 forcing to 2 ...
[   30.016345] rtl9300_force_sds_mode toggling LC or Ring for 10gr, round 0
[   30.076462] rtl9300_force_sds_mode end power 0x20 0 30
[   30.082203] rtl9300_force_sds_mode --------------------- serdes 6 forced to 2 DONE
[   30.091411] 8021q: adding VLAN 0 to HW filter on device lan9
[   30.098191] In rtl8226_advertise_aneg
[   30.111350] switch0: port 9(lan9) entered blocking state
[   30.117348] switch0: port 9(lan9) entered disabled state
[   30.124324] device lan9 entered promiscuous mode
[   30.139654] rtl83xx-switch lexra-bus0:switch@1b000000 lan10: configuring for phy/hsgmii link mode
[   30.149615] rtl930x_phylink_mac_config port 25, mode 0, phy-mode: hsgmii, speed -1, link 0
[   30.158906] rtl930x_phylink_mac_config SDS is 7
[   30.163984] rtl930x_phylink_mac_config: Unsupported speed: -1
[   30.170497] rtl930x_phylink_mac_config CURRENT FORCED MODE 18
[   30.177000] rtl9300_rtl8226_mode_set setting serdes 7 to mode sgmii +++++
[   30.216415] rtl9300_force_sds_mode: SDS: 7, mode 0
[   30.221769] rtl9300_force_sds_mode --------------------- serdes 7 forcing to 1f ...
[   30.230588] rtl9300_serdes_mac_link_config: registers before 00000000 00001403
[   30.238871] rtl9300_serdes_mac_link_config: registers after 00000000 00001403
[   30.249682] rtl9300_force_sds_mode: SDS: 7, mode 4
[   30.255034] rtl9300_force_sds_mode --------------------- serdes 7 forcing to 2 ...
[   30.476347] rtl9300_force_sds_mode toggling LC or Ring for 10gr, round 0
[   30.536493] rtl9300_force_sds_mode end power 0x20 0 30
[   30.542232] rtl9300_force_sds_mode --------------------- serdes 7 forced to 2 DONE
[   30.551361] 8021q: adding VLAN 0 to HW filter on device lan10
[   30.558236] In rtl8226_advertise_aneg
[   30.571655] switch0: port 10(lan10) entered blocking state
[   30.577859] switch0: port 10(lan10) entered disabled state
[   30.585067] device lan10 entered promiscuous mode
[   30.591541] rtl83xx_fib_event: FIB_RULE ADD/DEL for IPv6 not supported
[   30.599119] rtl83xx_fib_event: FIB_RULE ADD/DEL for IPv6 not supported
[   30.615443] rtl83xx-switch lexra-bus0:switch@1b000000 lan11: configuring for inband/1000base-x link mode
[   30.626161] rtl930x_phylink_mac_config port 26, mode 2, phy-mode: 1000base-x, speed -1, link 0
[   30.635898] rtl930x_phylink_mac_config SDS is 8
[   30.641271] rtl9300_configure_serdes CMU BAND is 16
[   30.646881] rtl9300_sds_set 31
[   30.686348] rtl9300_configure_serdes: Configuring RTL9300 SERDES 8
[   30.693321] rtl9300_serdes_mac_link_config: registers before 00000000 00001403
[   30.701612] rtl9300_serdes_mac_link_config: registers after 00000000 00001403
[   30.746347] rtl9300_force_sds_mode: SDS: 8, mode 0
[   30.751697] rtl9300_force_sds_mode --------------------- serdes 8 forcing to 1f ...
[   30.760456] rtl9300_force_sds_mode: SDS: 8, mode 22
[   30.765919] rtl9300_force_sds_mode --------------------- serdes 8 forcing to 4 ...
[   30.986345] rtl9300_force_sds_mode toggling LC or Ring for 10gr, round 0
[   31.076512] rtl9300_force_sds_mode end power 0x20 0 30
[   31.082253] rtl9300_force_sds_mode --------------------- serdes 8 forced to 4 DONE
[   31.100682] rtl9300_sds_10g_idle WARNING: Waiting for RX idle timed out, SDS 8
[   31.136406] start_1.2.1 ForegroundOffsetCal_Manual
[   31.141878] end_1.2.1
[   31.166405] start_1.2.3 Foreground Calibration
[   31.174229] rtl9300_do_rx_calibration_2_3: fgcal_gray: 31, fgcal_binary 31
[   31.181990] rtl9300_do_rx_calibration_2_3: end_1.2.3
[   31.187627] start_1.4.1
[   31.466363] end_1.4.1
[   31.469132] start_1.4.2
[   31.496407] vth_set_bin = 7
[   31.499149] vth_set_bin = 3
[   31.502246] Vth Maunal = 0
[   31.646400] Tap0 Sign : +
[   31.649432] tap0_coef_bin = 31
[   31.652335] tap0 manual = 0
[   31.655940] end_1.4.2
[   31.766724] rtl9300_sds_sym_err_get unsupported PHY-mode
[   31.796352] rtl9300_sds_sym_err_get unsupported PHY-mode
[   31.826409] start_1.2.1 ForegroundOffsetCal_Manual
[   31.831882] end_1.2.1
[   31.856407] start_1.2.3 Foreground Calibration
[   31.864227] rtl9300_do_rx_calibration_2_3: fgcal_gray: 31, fgcal_binary 31
[   31.871986] rtl9300_do_rx_calibration_2_3: end_1.2.3
[   31.877624] start_1.4.1
[   32.156346] end_1.4.1
[   32.159086] start_1.4.2
[   32.186400] vth_set_bin = 7
[   32.189169] vth_set_bin = 3
[   32.192280] Vth Maunal = 0
[   32.336397] Tap0 Sign : +
[   32.339430] tap0_coef_bin = 31
[   32.342333] tap0 manual = 0
[   32.345941] end_1.4.2
[   32.456731] rtl9300_sds_sym_err_get unsupported PHY-mode
[   32.486352] rtl9300_sds_sym_err_get unsupported PHY-mode
[   32.516420] start_1.2.1 ForegroundOffsetCal_Manual
[   32.521891] end_1.2.1
[   32.546486] start_1.2.3 Foreground Calibration
[   32.554327] rtl9300_do_rx_calibration_2_3: fgcal_gray: 31, fgcal_binary 31
[   32.562090] rtl9300_do_rx_calibration_2_3: end_1.2.3
[   32.567727] start_1.4.1
[   32.846347] end_1.4.1
[   32.849083] start_1.4.2
[   32.876407] vth_set_bin = 7
[   32.879145] vth_set_bin = 3
[   32.882241] Vth Maunal = 0
[   33.026399] Tap0 Sign : +
[   33.029427] tap0_coef_bin = 31
[   33.032330] tap0 manual = 0
[   33.035936] end_1.4.2
[   33.146864] rtl9300_sds_sym_err_get unsupported PHY-mode
[   33.176353] rtl9300_sds_sym_err_get unsupported PHY-mode
[   33.182281] rtl9300_configure_serdes CALIBTRATION FAILED
[   33.191364] rtl930x_phylink_mac_config: Unsupported speed: -1
[   33.198055] 8021q: adding VLAN 0 to HW filter on device lan11
[   33.205222] switch0: port 11(lan11) entered blocking state
[   33.211482] switch0: port 11(lan11) entered disabled state
[   33.218869] device lan11 entered promiscuous mode
[   33.235071] rtl83xx-switch lexra-bus0:switch@1b000000 lan12: configuring for inband/1000base-x link mode
[   33.245725] rtl930x_phylink_mac_config port 27, mode 2, phy-mode: 1000base-x, speed -1, link 0
[   33.255403] rtl930x_phylink_mac_config SDS is 9
[   33.260713] rtl9300_configure_serdes CMU BAND is 16
[   33.266164] rtl9300_sds_set 31
[   33.306372] rtl9300_configure_serdes: Configuring RTL9300 SERDES 9
[   33.313356] rtl9300_serdes_mac_link_config: registers before 00000000 00001403
[   33.321609] rtl9300_serdes_mac_link_config: registers after 00000000 00001403
[   33.366347] rtl9300_force_sds_mode: SDS: 9, mode 0
[   33.371700] rtl9300_force_sds_mode --------------------- serdes 9 forcing to 1f ...
[   33.380464] rtl9300_force_sds_mode: SDS: 9, mode 22
[   33.385924] rtl9300_force_sds_mode --------------------- serdes 9 forcing to 4 ...
[   33.470809] rtl930x_phylink_mac_config port 3, mode 0, phy-mode: xgmii, speed 1000, link 1
[   33.480072] rtl930x_phylink_mac_config SDS is -1
[   33.485236] rtl83xx-switch lexra-bus0:switch@1b000000 lan4: Link is Up - 1Gbps/Full - flow control rx/tx
[   33.495818] rtl930x_switch_irq link faults: 0fffff00
[   33.495851] rtl930x_switch_irq RX symbol errors: 00000004
[   33.507500] rtl930x_phylink_mac_config port 3, mode 0, phy-mode: xgmii, speed 1000, link 1
[   33.516785] rtl930x_phylink_mac_config SDS is -1
[   33.606364] rtl9300_force_sds_mode toggling LC or Ring for 10gr, round 0
[   33.696500] rtl9300_force_sds_mode end power 0x20 0 30
[   33.702240] rtl9300_force_sds_mode --------------------- serdes 9 forced to 4 DONE
[   33.721287] rtl9300_sds_10g_idle WARNING: Waiting for RX idle timed out, SDS 9
[   33.756407] start_1.2.1 ForegroundOffsetCal_Manual
[   33.761881] end_1.2.1
[   33.786396] start_1.2.3 Foreground Calibration
[   33.794264] rtl9300_do_rx_calibration_2_3: fgcal_gray: 31, fgcal_binary 31
[   33.802025] rtl9300_do_rx_calibration_2_3: end_1.2.3
[   33.807650] start_1.4.1
[   34.086344] end_1.4.1
[   34.089083] start_1.4.2
[   34.116433] vth_set_bin = 7
[   34.119180] vth_set_bin = 3
[   34.122276] Vth Maunal = 0
[   34.266436] Tap0 Sign : +
[   34.269483] tap0_coef_bin = 31
[   34.272402] tap0 manual = 0
[   34.276014] end_1.4.2
[   34.386731] rtl9300_sds_sym_err_get unsupported PHY-mode
[   34.416353] rtl9300_sds_sym_err_get unsupported PHY-mode
[   34.446409] start_1.2.1 ForegroundOffsetCal_Manual
[   34.451880] end_1.2.1
[   34.476410] start_1.2.3 Foreground Calibration
[   34.484235] rtl9300_do_rx_calibration_2_3: fgcal_gray: 31, fgcal_binary 31
[   34.491988] rtl9300_do_rx_calibration_2_3: end_1.2.3
[   34.497612] start_1.4.1
[   34.776350] end_1.4.1
[   34.779089] start_1.4.2
[   34.806409] vth_set_bin = 7
[   34.809151] vth_set_bin = 3
[   34.812247] Vth Maunal = 0
[   34.956397] Tap0 Sign : +
[   34.959424] tap0_coef_bin = 31
[   34.962327] tap0 manual = 0
[   34.965937] end_1.4.2
[   35.076748] rtl9300_sds_sym_err_get unsupported PHY-mode
[   35.106356] rtl9300_sds_sym_err_get unsupported PHY-mode
[   35.136422] start_1.2.1 ForegroundOffsetCal_Manual
[   35.141896] end_1.2.1
[   35.166409] start_1.2.3 Foreground Calibration
[   35.174239] rtl9300_do_rx_calibration_2_3: fgcal_gray: 31, fgcal_binary 31
[   35.181987] rtl9300_do_rx_calibration_2_3: end_1.2.3
[   35.187609] start_1.4.1
[   35.466484] end_1.4.1
[   35.469228] start_1.4.2
[   35.496409] vth_set_bin = 7
[   35.499149] vth_set_bin = 3
[   35.502244] Vth Maunal = 0
[   35.646419] Tap0 Sign : +
[   35.649448] tap0_coef_bin = 31
[   35.652351] tap0 manual = 0
[   35.655962] end_1.4.2
[   35.766732] rtl9300_sds_sym_err_get unsupported PHY-mode
[   35.796365] rtl9300_sds_sym_err_get unsupported PHY-mode
[   35.802294] rtl9300_configure_serdes CALIBTRATION FAILED
[   35.811296] rtl930x_phylink_mac_config: Unsupported speed: -1
[   35.817985] 8021q: adding VLAN 0 to HW filter on device lan12
[   35.824689] switch0: port 4(lan4) entered blocking state
[   35.830732] switch0: port 4(lan4) entered forwarding state
[   35.837378] IPv6: ADDRCONF(NETDEV_CHANGE): switch0: link becomes ready
[   35.845241] switch0: port 12(lan12) entered blocking state
[   35.851500] switch0: port 12(lan12) entered disabled state
[   35.859108] device lan12 entered promiscuous mode
[   35.865045] IPv6: ADDRCONF(NETDEV_CHANGE): switch0.1: link becomes ready
[   35.897981] rtl83xx_fib_event: FIB_RULE ADD/DEL for IPv6 not supported
[   37.546464] rtl83xx_fib_event: FIB_RULE ADD/DEL for IPv6 not supported
[   40.099771] rtl930x_set_l3_egress_intf writing to index 0: 00000201 10101492
[   40.109482] Not offloading default route for now
[ 1542.774521] rtl930x_phylink_mac_config port 25, mode 0, phy-mode: hsgmii, speed 2500, link 1
[ 1542.783984] rtl930x_phylink_mac_config SDS is 7
[ 1542.789117] rtl930x_phylink_mac_config CURRENT FORCED MODE 2
[ 1542.795447] rtl9300_rtl8226_mode_set setting serdes 7 to mode hsgmii +++++
[ 1542.838614] rtl9300_force_sds_mode: SDS: 7, mode 0
[ 1542.843971] rtl9300_force_sds_mode --------------------- serdes 7 forcing to 1f ...
[ 1542.852757] rtl9300_serdes_mac_link_config: registers before 00000000 00001403
[ 1542.861030] rtl9300_serdes_mac_link_config: registers after 00000000 00001403
[ 1542.871404] rtl9300_force_sds_mode: SDS: 7, mode 18
[ 1542.876850] rtl9300_force_sds_mode --------------------- serdes 7 forcing to 12 ...
[ 1542.879819] rtl930x_switch_irq link faults: 0fffff00
[ 1542.890891] rtl930x_switch_irq RX symbol errors: 000000c4
[ 1543.098642] rtl9300_force_sds_mode toggling LC or Ring for 10gr, round 0
[ 1543.158728] rtl9300_force_sds_mode end power 0x20 0 30
[ 1543.164468] rtl9300_force_sds_mode --------------------- serdes 7 forced to 12 DONE
[ 1543.173429] rtl83xx-switch lexra-bus0:switch@1b000000 lan10: Link is Up - 2.5Gbps/Full - flow control off
[ 1543.184207] rtl930x_phylink_mac_config port 25, mode 0, phy-mode: hsgmii, speed 2500, link 1
[ 1543.193676] rtl930x_phylink_mac_config SDS is 7
[ 1543.198823] rtl930x_phylink_mac_config CURRENT FORCED MODE 18
[ 1543.205287] switch0: port 10(lan10) entered blocking state
[ 1543.211472] switch0: port 10(lan10) entered forwarding state
[ 1543.331411] rtl930x_phylink_mac_config port 24, mode 0, phy-mode: hsgmii, speed 2500, link 1
[ 1543.340879] rtl930x_phylink_mac_config SDS is 6
[ 1543.345927] rtl930x_phylink_mac_config CURRENT FORCED MODE 2
[ 1543.352320] rtl9300_rtl8226_mode_set setting serdes 6 to mode hsgmii +++++
[ 1543.398619] rtl9300_force_sds_mode: SDS: 6, mode 0
[ 1543.403971] rtl9300_force_sds_mode --------------------- serdes 6 forcing to 1f ...
[ 1543.412981] rtl9300_serdes_mac_link_config: registers before 00000000 00001403
[ 1543.421357] rtl9300_serdes_mac_link_config: registers after 00000000 00001403
[ 1543.432789] rtl9300_force_sds_mode: SDS: 6, mode 18
[ 1543.438245] rtl9300_force_sds_mode --------------------- serdes 6 forcing to 12 ...
[ 1543.439817] rtl930x_switch_irq link faults: 0fffff00
[ 1543.452286] rtl930x_switch_irq RX symbol errors: 000000c4
[ 1543.658628] rtl9300_force_sds_mode toggling LC or Ring for 10gr, round 0
[ 1543.718719] rtl9300_force_sds_mode end power 0x20 0 30
[ 1543.724459] rtl9300_force_sds_mode --------------------- serdes 6 forced to 12 DONE
[ 1543.733780] rtl930x_switch_irq link faults: 0dffff00
[ 1543.733825] rtl83xx-switch lexra-bus0:switch@1b000000 lan9: Link is Up - 2.5Gbps/Full - flow control off
[ 1543.739346] rtl930x_switch_irq RX symbol errors: 000000c4
[ 1543.756120] rtl930x_phylink_mac_config port 24, mode 0, phy-mode: hsgmii, speed 2500, link 1
[ 1543.765603] rtl930x_phylink_mac_config SDS is 6
[ 1543.770761] rtl930x_phylink_mac_config CURRENT FORCED MODE 18
[ 1543.777310] switch0: port 9(lan9) entered blocking state
[ 1543.783337] switch0: port 9(lan9) entered forwarding state
[ 1547.973386] rtl930x_switch_irq link faults: 0cffff00
[ 1547.973396] rtl83xx-switch lexra-bus0:switch@1b000000 lan10: Link is Down
[ 1547.973422] rtl930x_switch_irq RX symbol errors: 000000c4
[ 1547.992849] switch0: port 10(lan10) entered disabled state
[ 1548.529704] rtl930x_switch_irq link faults: 0cffff00
[ 1548.535256] rtl930x_switch_irq RX symbol errors: 000000c4
[ 1548.541357] rtl83xx-switch lexra-bus0:switch@1b000000 lan9: Link is Down
[ 1548.549012] switch0: port 9(lan9) entered disabled state
```
</details>

oliv3r commented:
> Booting the image that is created from your dev/realtek-wip branch on your fork will panic but not koreapyj repo his image is the only one I can get to boot and that works

Could be that it's a non-working state right now (wip being WIP and me often pushing there, sorry for that).

Can you paste the crash-log so I know what's pukin' and I can fix it?

oliv3r commented:
Thanks, but the earlier log is still on 5.15.104, where we are on 5.15.106; so no idea how old that is. Also, i still can't believe it rashes on the serial output ... I mean, that's not even code any of us has touched in ages.

oliv3r commented:
So I just booted my current WIP tree a few times, and it boots fine; can you build and compile it? if it fails again, double check the RAM size, as that's the only thing I can imagine causing these weird things.

koreapyj commented:
It "was" unmanaged switch but it seems like it's now managed one. http://hasivo.com/productinfo/900908.html

hailfinger commented:
@koreapyj is there a chance you could rebase this to the current tree? Thanks.

Djfe commented:
A very similar switch was just merged last week.
Feel free to update your PR and maybe create a common .dtsi if it makes sense (for dts parts that are shared between both variants: the F1100W and the S1100W)
https://github.com/openwrt/openwrt/commit/ed7d62caf27d41b9aa7b0426f5acd2e40e61285d

koreapyj commented:
I know rtl83xx target have big positive changes but I lost serial TX for unknown reason. I'll try to fix it but currently I don't know I can fix it or not.

Djfe commented:
maybe your usb tty adapter has a fake chip. I used to have that problem (though tx didn't work from the get go)
http://www.starlino.com/ftdi-chip-real-of-fake-how-to-spot-a-fake-rt232r-rt232rl-and-others.html
